patches/tsc-multi@1.1.0.patch

```diff
@@ -1,96 +1,196 @@
+diff --git a/README.md b/README.md
+index 419b3c2f5505ea41cbce33edaea33b1c02c2f609..903f3b85c816f874342dd97d5b90aebe43d73891 100644
+--- a/README.md
++++ b/README.md
+@@ -72,12 +72,16 @@ tsc-multi --clean
+ 
+ ### `targets`
+ 
+-Build targets. All options except `extname` will override `compilerOptions` in `tsconfig.json`.
++Build targets. All options except `extname` and `packageOverrides` will override `compilerOptions` in `tsconfig.json`.
+ 
+ ```js
+ {
+   // Rename the extension of output files
+   extname: ".js",
++  // Set the output package type to module
++  packageOverrides: {
++    "package.json": { "type": "module" }
++  },
+   // Skip type-checking (Experimental)
+   transpileOnly: false,
+   // Compiler options
 diff --git a/dist/build.js b/dist/build.js
-index 29144bc5cdb118655229a1f8998c3168170ca257..19b3d6b3066d308d0eb53dfae65fe4806f45176b 100644
+index 29144bc5cdb118655229a1f8998c3168170ca257..575e71dfae7b459e327b8f8f9c016aaeff7e5cb2 100644
 --- a/dist/build.js
 +++ b/dist/build.js
-@@ -56,7 +56,7 @@ async function build({ targets: inputTargets, stdout = "inherit", stderr = "inhe
+@@ -13,18 +13,26 @@ const debug_1 = tslib_1.__importDefault(require("./debug"));
+ const WORKER_PATH = (0, path_1.join)(__dirname, "worker/entry.js");
+ const DEFAULT_EXTNAME = ".js";
+ function validateTargets(targets) {
+-    const extnames = targets.map((target) => target.extname || DEFAULT_EXTNAME);
+-    const extMap = new Map();
+-    for (let i = 0; i < extnames.length; i++) {
+-        const ext = extnames[i];
+-        if (!ext.startsWith(".")) {
++    const outputDifferentiation = targets.map((target) => `${target.extname || DEFAULT_EXTNAME}+${target.outDir ?? "<tscfg-outdir>"}`);
++    const outputMap = new Map();
++    for (let i = 0; i < outputDifferentiation.length; i++) {
++        const uniqueOutput = outputDifferentiation[i];
++        if (!uniqueOutput.startsWith(".")) {
+             throw new Error(`targets[${i}].extname must be started with ".".`);
+         }
+-        const existedIndex = extMap.get(ext);
+-        if (existedIndex != null) {
+-            throw new Error(`targets[${i}].extname is already used in targets[${existedIndex}].extname`);
++        const existedIndex = outputMap.get(uniqueOutput);
++        if (existedIndex !== undefined) {
++            throw new Error(`targets[${i}].extname and/or .outDir is already used in targets[${existedIndex.index}]`);
++        }
++        outputMap.set(uniqueOutput, { index: i, outDir: targets[i].outDir });
++        const packageOverrides = targets[i].packageOverrides;
++        if (packageOverrides) {
++            Object.keys(packageOverrides).forEach((packageName) => {
++                if (!packageName.endsWith("package.json")) {
++                    throw new Error(`targets[${i}].packageOverrides[${packageName}] may only reference "package.json" paths`);
++                }
++            });
+         }
+-        extMap.set(ext, i);
```

Comment:
Pre-existing: Looks reasonable, but a few comments explaining what is being tested/rejected would help.

// Ensure that ...
const uniqueOutput = ...

// Ensure that ...
const existedIndex = ...

...and so on.